### PR TITLE
Remove `lsm-tree-utils` sub-library

### PR DESCRIPTION
**Pull request by jorisdral**

### CODE REVIEW

File: lsm-tree.cabal [hunk 1]
```diff
@@ -289,20 +268,25 @@ test-suite lsm-tree-test
...
+    , wide-word
 
   ghc-options:      -fno-ignore-asserts -threaded
 
 benchmark lsm-tree-micro-bench
   import:           warnings, wno-x-partial
   default-language: Haskell2010
   type:             exitcode-stdio-1.0
-  hs-source-dirs:   bench/micro
+  hs-source-dirs:   bench/micro test
```
Comment (phadej) [inline lsm-tree.cabal:279]: So now the `Extras` are build multiple times for different components, `lsm-tree-test`, `lsm-tree-micro-bench` and also `lsm-tree-macro-bench`?

Comment (mheinzel) [reply lsm-tree.cabal:279]: As I understood it the idea was to import the test suite, but then I realised that's not possible (or is it?). In that case, I'm not sure removing `utils` is worth it. :thinking:

Comment (dcoutts) [reply lsm-tree.cabal:279]: Yes the small number of modules used (out of the small number of util/extra modules that there are) would be built for each component that uses them. My view was that was a reasonable tradeoff to eliminate a whole sub-heirarchy.

**Review by dcoutts — APPROVED** [review]
:+1:

**Pull request merged**